content/transfer-hook.md

```diff
@@ -0,0 +1,2918 @@
+
+      payerWallet.publicKey, // mint authority
+
+      null, // freeze authority
+
+      TOKEN_2022_PROGRAM_ID,
+
+    ),
+
+    createInitializeInstruction({
+
+      programId: TOKEN_2022_PROGRAM_ID,
+
+      mint: cookieMint.publicKey,
+
+      metadata: cookieMint.publicKey,
+
+      name: cookieMetadata.name,
+
+      symbol: cookieMetadata.symbol,
+
+      uri: cookieMetadata.uri,
+
+      mintAuthority: payerWallet.publicKey,
+
+      updateAuthority: payerWallet.publicKey,
+
+    }),
+
+    createAssociatedTokenAccountInstruction(
+
+      payerWallet.publicKey, // payer
+
+      sourceCookieAccount, // associated token account
+
+      payerWallet.publicKey, // owner
+
+      cookieMint.publicKey, // mint
+
+      TOKEN_2022_PROGRAM_ID,
+
+    ),
+
+    createMintToInstruction(
+
+      cookieMint.publicKey, // mint
+
+      sourceCookieAccount, // destination
+
+      payerWallet.publicKey, // authority
+
+      1, // amount - NFTs there will only be one
+
+      [], // multi signers
+
+      TOKEN_2022_PROGRAM_ID,
+
+    ),
+
+    createSetAuthorityInstruction(
+
+      // revoke mint authority
+
+      cookieMint.publicKey, // mint
+
+      payerWallet.publicKey, // current authority
+
+      AuthorityType.MintTokens, // authority type
+
+      null, // new authority
+
+      [], // multi signers
+
+      TOKEN_2022_PROGRAM_ID,
+
+    ),
+
+  );
+
+
+
+  const txSig = await sendAndConfirmTransaction(connection, transaction, [payerWallet.payer, cookieMint]);
+
+  console.log(getExplorerLink('transaction', txSig, 'localnet'));
+
+});
+
+```
+
+
+
+Feel free to run the first test to make sure everything is working:
+
+```bash
+
+anchor test
+
+```
+
+
+
+### 4. Write the "Create Crumb Mint" test
+
+
+
+Now that we have our cookie NFT, we need our crumb SFTs. Creating the crumbs that will be minted on each transfer of our cookie will be our second test.
+
+
+
+Remember our crumbs are a Token Program mint, and to attach metadata we need to use Metaplex.
+
+
+
+First, we need to grab some Metaplex accounts and format our metadata.
+
+
+
+To format our metadata, we need to satisfy Metaplex's `DataV2` struct - for this, we only need to append some additional fields to our `crumbMetadata`.
+
+
+
+The Metaplex accounts we will need are:
+
+- `TOKEN_METADATA_PROGRAM_ID`: The Metaplex program
+
+- `metadataPDA`: The metadata account PDA derived from our `crumbMint`
+
+
+
+Lastly, to create our crumb, we need the following instructions:
+
+- `SystemProgram.createAccount`: Saves space for our mint
+
+- `createInitializeMintInstruction`: Initializes our mint
+
+- `createCreateMetadataAccountV3Instruction`: Creates the metadata account
+
+- `createSetAuthorityInstruction`: This sets the mint authority to the `crumbMintAuthority`, which is the PDA our transfer hook program owns
+
+
+
+Putting it all together we get the following:
+
+
+
+```ts
+
+  it('Create Crumb Mint', async () => {
+
+    // SFT Should have 0 decimals
```

Comment:
I thought SFTs have more than 0 decimals?

Reply:
Not necessarily, items in games don't have decimals - but are SFTs. Think of it like a NFT with more than one print